service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/CreateAiNotebookInstanceStep.java

```diff
@@ -223,6 +219,17 @@ static Instance setFields(
               .setRepository(containerImageParameters.getRepository())
               .setTag(containerImageParameters.getTag()));
     }
+    // Set metadata
+    Map<String, String> metadata = new HashMap<>();
+    if (containerImageParameters != null) {
+      // User needs to run as Jupyter instead of root to have the post-startup.sh run successfully.
+      // This is not a TERRA reserved metadata key, so it is overridable. If the creationParameter
+      // says otherwise, we will set the value to what the user specified.
+      metadata.put(NOTEBOOK_DISABLE_ROOT_METADATA_KEY, "true");
+    }
+    Optional.ofNullable(creationParameters.getMetadata()).ifPresent(metadata::putAll);
```

Comment:
What stops a user from setting NOTEBOOK_DISABLE_ROOT_METADATA_KEY="false" in their metadata? I think it would overwrite the setting above and give root access on a custom image.

Reply:
Why add the defaulting code at all then? Just let the caller specify it if they want to.

Clearly, I don't understand the requirements or the security philosophy of this change.